File: python/search/clustering.py
from pymilvus import MilvusClient, DataType

client = MilvusClient(
    uri="http://localhost:19530",
    token="root:Milvus"
)

schema = client.create_schema()
schema.add_field(field_name="id", datatype=DataType.INT64, is_primary=True)
schema.add_field(field_name="vector", datatype=DataType.FLOAT_VECTOR, dim=5)

client.drop_collection(collection_name="my_collection")
client.create_collection(
    collection_name="my_collection",
    schema=schema,
    clustering_key_field=True,
)

from pymilvus import connections, Collection

connections.connect(
    uri="http://localhost:19530",
    token="root:Milvus"
)

collection = Collection("my_collection")

collection.compact(is_clustering=True)

if collection.wait_for_compaction_completed(is_clustering=True):
    collection.get_compaction_state(is_clustering=True)

File: java/src/main/java/org/example/storage/Clustering.java
package org.example.storage;

import io.milvus.v2.client.ConnectConfig;
import io.milvus.v2.client.MilvusClientV2;
import io.milvus.v2.common.DataType;
import io.milvus.v2.service.collection.request.AddFieldReq;
import io.milvus.v2.service.collection.request.CreateCollectionReq;
import io.milvus.v2.service.collection.request.DropCollectionReq;
import io.milvus.v2.service.utility.request.CompactReq;
import io.milvus.v2.service.utility.request.GetCompactionStateReq;
import io.milvus.v2.service.utility.response.CompactResp;
import io.milvus.v2.service.utility.response.GetCompactionStateResp;

public class Clustering {
    private static final MilvusClientV2 client;
    static {
        client = new MilvusClientV2(ConnectConfig.builder()
                .uri("http://localhost:19530")
                .token("root:Milvus")
                .build());
    }

    public static void main(String[] args) {
        CreateCollectionReq.CollectionSchema schema = client.createSchema();

        schema.addField(AddFieldReq.builder()
                .fieldName("id")
                .dataType(DataType.Int64)
                .isPrimaryKey(true)
                .autoID(false)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("key")
                .dataType(DataType.Int64)
                .isClusteringKey(true)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("var")
                .dataType(DataType.VarChar)
                .maxLength(1000)
                .build());

        schema.addField(AddFieldReq.builder()
                .fieldName("vector")
                .dataType(DataType.FloatVector)
                .dimension(5)
                .build());

        client.dropCollection(DropCollectionReq.builder()
                .collectionName("clustering_test")
                .build());

        CreateCollectionReq requestCreate = CreateCollectionReq.builder()
                .collectionName("clustering_test")
                .collectionSchema(schema)
                .build();
        client.createCollection(requestCreate);

        CompactResp compactResp = client.compact(CompactReq.builder()
                .collectionName("clustering_test")
                .isClustering(true)
                .build());

        GetCompactionStateResp stateResp = client.getCompactionState(GetCompactionStateReq.builder()
                .compactionID(compactResp.getCompactionID())
                .build());

        System.out.println(stateResp.getState());
    }
}
